src/rpcq-python.lisp

```diff
@@ -0,0 +1,287 @@
+;;;; rpcq-python.lisp
+;;;;
+;;;; Authors: Nikolas Tezak, Eric Peterson
+
+(in-package #:rpcq)
+
+
+(defparameter *python-types*
+  '(:string "str"
+    :bytes "bytes"
+    :float "float"
+    :integer "int"
+    :bool "bool"
+    :map "Dict"
+    :list "List"
+    :any "Any"))
+
+(defparameter *python-instance-check-types*
+  '(:string "basestring"
+    :bytes "bytes"
+    :float "float"
+    :integer "int"
+    :bool "bool"
+    :map "dict"
+    :list "list"
+    :any "object"))
+
+
+(defun python-instance-check-type (field-type)
+  (let ((*python-types* *python-instance-check-types*))
+    (python-type field-type)))
+
+(defun python-argspec-default (field-type default)
+  "Translate DEFAULT values for immutable objects of a given
+FIELD-TYPE to python."
+  (case field-type
+    ((:string)
+     (if default
+         (format nil "~S" default)
+         "None"))
+    ((:bytes)
+     (if default
+         (format nil "b~S" (to-string default))
+         "None"))
+    ((:bool)
+     (if default
+         "True"
+         "False"))
+    ((:integer)
+     (if default
+         (format nil "~d" default)
+         "None"))
+    ((:float)
+     (if default
+         (format nil "~e" default)
+         "None"))
+    (otherwise
+     (cond
```

Comment:
A construction `(cons sym x)` has the type `(cons (eql sym) *)` which can be shortened to `(cons (eql sym))`, and it works with `typecase`

```
(typecase field-type
  ((cons (eql :list)) ...)
  ((cons (eql :map)) ...)
  (otherwise "None"))
```

Reply:
whoa this is cool, ty